Source/Layout/ASLayoutElement.mm

```diff
@@ -180,93 +180,91 @@ - (instancetype)init
 
 - (ASLayoutElementSize)size
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  return _size;
+  return _size.load();
 }
 
 - (void)setSize:(ASLayoutElementSize)size
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  _size = size;
+  _size.store(size);
 }
 
 #pragma mark - ASLayoutElementStyleSizeForwarding
 
 - (ASDimension)width
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  return _size.width;
+  return _size.load().width;
 }
 
 - (void)setWidth:(ASDimension)width
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  _size.width = width;
+  ASLayoutElementSize newSize = _size.load();
+  newSize.width = width;
+  _size.store(newSize);
   ASLayoutElementStyleCallDelegate(ASLayoutElementStyleWidthProperty);
 }
 
 - (ASDimension)height
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  return _size.height;
+  return _size.load().height;
 }
 
 - (void)setHeight:(ASDimension)height
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  _size.height = height;
+  ASLayoutElementSize newSize = _size.load();
+  newSize.height = height;
+  _size.store(newSize);
```

Comment:
Unfortunately this breaks atomicity =/ Imagine:

Thread 1: read size A
Thread 2: read size A
Thread 1: write size { A, height = x }
Thread 2: write size { A, width = y }

So we end up with `size { A, width = y }`, thread 2 overwrote thread 1's change. We should have ended up with `size { A, height = x, width = y }`.

We have to do the read-update-write all in one operation.